### PR TITLE
Feat/adding chatapp message delete frontend

### PR DESCRIPTION
**Pull request by arnb-smnta**
Backend Part-Had to fix the socketEnum(delete message event) typo in backend and change the status codes because 401 status code was logging out user in frontend.

Frontend Part-Added the buttons only for own message of users for deleting the message might have to add options for both types of messages in later parts when more functionality is introduced , Implemented different styling for content and attachment types of messages .

on deletion-last message of chat is updated along with the current chat of the user is also updated.

small demo of UI & functionality- <a href="https://res.cloudinary.com/dxfdxwvlm/video/upload/v1712417862/chat-app_owtwtw.mkv">Video Link 1</a>
<a href="https://res.cloudinary.com/dxfdxwvlm/video/upload/v1712422449/2024-04-06_22-21-13_n6gahr.mkv" >Video link 2</a>

dont mind the sounds please

### CODE REVIEW

**Comment by arnb-smnta** [comment]
any updates sir

**Comment by arnb-smnta** [comment]
any updates sir

File: src/controllers/apps/chat-app/message.controllers.js [hunk 1]
```diff
@@ -186,8 +186,8 @@ const deleteMessage = asyncHandler(async (req, res) => {
   // Check if user is the sender of the message
   if (message.sender.toString() !== req.user._id.toString()) {
     throw new ApiError(
-      401,
-      "You are not authorised to delete the message ,you are not the sender"
+      400,
+      "You are not the owner to delete the message ,you are not the sender"
```
Comment (wajeshubham) [inline src/controllers/apps/chat-app/message.controllers.js:190]: Fix the formatting there should be space after the comma and not before and the error code should be 403.

**Review by wajeshubham — CHANGES_REQUESTED** [review]

File: src/controllers/apps/chat-app/message.controllers.js (hunk 1 above)
Comment (itsyashi126) [reply src/controllers/apps/chat-app/message.controllers.js:190]: Who's code is this yours

Comment (arnb-smnta) [reply src/controllers/apps/chat-app/message.controllers.js]: yes i have written the additional code is something wrong

**Review by wajeshubham — APPROVED** [review]

**Pull request merged**